src/rolling/file_number.rs

```diff
@@ -1,67 +1,94 @@
-use std::cmp::Reverse;
-use std::fmt;
-use std::sync::{Arc, Mutex};
+use std::collections::BTreeSet;
+use std::sync::Arc;
 
-#[derive(Clone, Default)]
-pub struct FileNumber {
-    inner: Arc<Inner>,
+pub struct FileTracker {
+    files: BTreeSet<FileNumber>,
 }
 
-#[derive(Default)]
-struct Inner {
-    pub(crate) file_number: u32,
-    next_file_number: Arc<Mutex<Option<FileNumber>>>,
-}
+impl FileTracker {
+    pub fn new() -> FileTracker {
+        FileTracker::from_file_numbers(vec![0]).unwrap()
+    }
 
-impl fmt::Debug for FileNumber {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.debug_struct("FileNumber")
-            .field("file_number", &self.inner.file_number)
-            .finish()
+    pub fn first(&self) -> &FileNumber {
+        self.files.iter().next().unwrap()
     }
-}
 
-impl Ord for FileNumber {
-    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
-        self.inner.file_number.cmp(&other.inner.file_number)
+    pub fn take_first_unused(&mut self) -> Option<FileNumber> {
```

Comment:
A comment over this method would be nice. The semantic are not aligned with the name.

For instance... the corner case of `self.files.len == 1`.
And also, 

if we have file number 1, 2, 3...
And 1 is used but 2 is unused, we do not return 2. (that's important)